common/dbconfig.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#include "common/dbconfig.h"
+#include "common/segment_utils.h"
+#include <utility>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+DECLARE_int32(replicator_replication_mode);
+
+namespace common {
+
+DBConfigManager* DBConfigManager::get() {
+  static DBConfigManager instance;
+  return &instance;
+}
+
+DBConfigManager::DBConfigManager() : dbConfig(new DBConfig()) {
+}
+
+bool DBConfigManager::loadJsonStr(const std::string& content) {
+  Json::Value root;
+  Json::Reader reader;
+  if (!reader.parse(content, root) || !root.isObject()) {
```

Comment:
nit: check `!root.isObject()` first?

Reply:
`isObject` refers  to check if the json in `{..}` (Json Object) format & not an array `[..]` or just bool like `name:prem` .. So root will be filled after parsing. Without parsing. root.isNull() will be true.

Reply:
ah sorry, I confused myself